modules/datahandling/datahandler.go

```diff
@@ -14,5 +18,40 @@ type DataHandler struct {
 // processing the data, and updating DB/FS/RabbitMQ as needed.
 func (dh DataHandler) Handle(wsID uint64, messageType int, message []byte) error {
 	fmt.Printf("Handling Message: %s\n", message)
+
+	req, err := createAbstractRequest(message)
+	if err != nil {
+		utils.LogOnError(err, "Failed to parse json")
+		return err
+	}
+
+	// automatically delegates if the request is authenticated or not
+	fullR, err := getFullRequest(req)
+
+	if err != nil {
+		utils.LogOnError(err, "Failed to construct full request")
+		return err
+	}
+
+	response, notification, err := fullR.process()
+
+	if err != nil {
+		utils.LogOnError(err, "Failed to process request")
+		return err
+	}
+
+	if response != nil {
+		// TODO: send on rabbit
+	}
+
+	if notification != nil {
+		// TODO: send on rabbit
+	}
+
 	return nil
```

Comment:
Are there any tests for datahandler? it's arguably one of the most important parts of the backend.